back/app/Http/Controllers/WeaponController.php

```diff
@@ -0,0 +1,94 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use Illuminate\Http\Request;
+use App\Models\Weapon;
+use App\Http\Requests\WeaponIndexRequest;
+use App\Http\Requests\WeaponStoreRequest;
+use Illuminate\Support\Facades\Storage;
+use Illuminate\Support\Facades\DB;
+use Exception;
+
+class WeaponController extends Controller
+{
+    public function index(WeaponIndexRequest $request)
+    {
+        try{
+            $validated = $request->validated();
+            $query = Weapon::query();
+            //検索
+            if (!empty($validated['name'])) {
+                $query->where('name', 'like', '%' . $validated['name'] . '%');
+            }
+            //絞り込み
+            if (isset($validated['physicsType'])) {
+                $query->where('physics_type', $validated['physicsType']);
+            }
+
+            if (isset($validated['elementType'])) {
+                $query->where('element_type', $validated['elementType']);
+            }
+            //並び替え
+            $sortOptions = [
+                'createdAt' => 'created_at',
+                'updatedAt' => 'updated_at',
+                'physicsAttack' => 'physics_attack',
+                'elementType' => 'element_type',
+                'name' => 'name',
+            ];
+            $sortColumn = $sortOptions[$validated['sort']];
+            $sortDirection = $validated['desc'] ? 'desc' : 'asc';
+            $query->orderBy($sortColumn, $sortDirection);
+            //ページネーション
+            $currentPage = $validated['currentPage'] ?? 1;
+            $weapons = $query->paginate(40, ['*'], 'page', $currentPage);
+            $formattedWeapons = $weapons->map(function ($weapon) {
+                return[
+                    'id' => $weapon->id,
+                    'name' => $weapon->name,
+                    'imageUrl' => $weapon->image_url,
+                    'physicsType' => $weapon->physics_type,
+                    'elementType' => $weapon->element_type,
+                ];
+            });
+            return response()->json([
+                'weapons' => $formattedWeapons
+            ]);
```

Comment:
jsonで直接返すのと、合計ページをHeaderで返してください

```suggestion
            $formattedWeapons = $weapons->map(function ($weapon) {
                return[
                    'id' => $weapon->id,
                    'name' => $weapon->name,
                    'imageUrl' => $weapon->image_url,
                    'physicsType' => $weapon->physics_type,
                    'elementType' => $weapon->element_type,
                ];
            });
            return response()->json($formattedWeapons)
              ->header('X-TOTAL-PAGE', $totalPage);
```